libcloud/compute/drivers/outscale.py

```diff
@@ -0,0 +1,1018 @@
+            data.update({"PublicIp": public_ip})
+        if public_ip_id is not None:
+            data.update({"PublicIpId": public_ip_id})
+        if nic_id is not None:
+            data.update({"NicId": nic_id})
+        if vm_id is not None:
+            data.update({"VmId": vm_id})
+        if allow_relink is not None:
+            data.update({"AllowRelink": allow_relink})
+        data = json.dumps(data)
+        if self._call_api(action, data).status_code == 200:
+            return True
+        return False
+
+    def ex_detach_public_ip(self,
+                            public_ip: str = None,
+                            link_public_ip_id: str = None,
+                            dry_run: bool = False):
+        """
+        Detach a volume.
+
+        :param      public_ip: (Required in a Net) The ID representing the
+        association of the EIP with the VM or the NIC
+        :type       public_ip: ``str``
+
+        :param      link_public_ip_id: (Required in a Net) The ID
+        representing the association of the EIP with the
+        VM or the NIC.
+        :type       link_public_ip_id: ``str``
+
+        :param      dry_run: If true, checks whether you have the required
+        permissions to perform the action.
+        :type       dry_run: ``bool``
+
+        :return: the attached volume
+        :rtype: ``dict``
+        """
+        action = "UnlinkPublicIp"
+        data = {"DryRun": dry_run}
+        if public_ip is not None:
+            data.update({"PublicIp": public_ip})
+        if link_public_ip_id is not None:
+            data.update({"LinkPublicIpId": link_public_ip_id})
+        data = json.dumps(data)
+        if self._call_api(action, data).status_code == 200:
+            return True
+        return False
+
+    def create_node(self,
+                    ex_image_id: str,
+                    ex_dry_run: bool = False,
+                    ex_block_device_mapping: dict = None,
+                    ex_boot_on_creation: bool = True,
+                    ex_bsu_optimized: bool = True,
+                    ex_client_token: str = None,
+                    ex_deletion_protection: bool = False,
+                    ex_keypair_name: str = None,
+                    ex_max_vms_count: int = None,
+                    ex_min_vms_count: int = None,
+                    ex_nics: dict = None,
+                    ex_performance: str = None,
+                    ex_placement: dict = None,
+                    ex_private_ips: [str] = None,
+                    ex_security_group_ids: [str] = None,
+                    ex_security_groups: [str] = None,
+                    ex_subnet_id: str = None,
+                    ex_user_data: str = None,
+                    ex_vm_initiated_shutdown_behavior: str = None,
+                    ex_vm_type: str = None
+                    ):
+        """
+        Create a new instance.
+
+        :param      ex_image_id: The ID of the OMI used to create the VM.
```

Comment:
That's fine for now, but ideally this method would comply with the standard API in the future - this means take name, image and size arguments.

This would make it much easier for users to switch from other providers to this one without needing to change much code.